### PR TITLE
Add support for listing operators in limitation clauses

### LINKED ISSUE #17003: Support for operators in forwarding only/except lists, and in imports and only/except lists for use statements
When forwarding a type that has an operator method, it seems obvious that users would want a way to explicitly opt in or out of forwarding operator methods defined on the type.  For instance,

```chapel
record Container {
  var x: OtherType;

  forwarding var only <; // or `forwarding var except +`
}

record OtherType {
  var y: int;
}
proc type OtherType.+(lhs: OtherType, rhs: OtherType) {
  return new OtherType(lhs.y + rhs.y);
}

proc type OtherType.<(lhs: OtherType, rhs: OtherType) {
  return lhs.y < rhs.y;
}

```

It seems reasonable to also add the support to use/import statements, since the only/except lists are similar to those used with forwarding.

Note that someone coming from Rust or Python might think `import M.*;` would mean "import everything from M", so it would be good to generate a warning for this case (that's potentially suppressible with a flag) along the lines of "Note that 'import M.*;' requests 'operator *' to be imported from module M in Chapel; if you intended to import all symbols from 'M', please use 'use M;' instead" or potentially only allow operators to be listed in the curly bracket import form, e.g.

```chapel
import M.{*};
```

### PR DESCRIPTION
**Pull request by lydia-duncan**
Prior to this work, listing operators in `only`, `except` or `import` clauses
resulted in a syntax error.  This change makes it so that operators can
be listed in the `only` or `except` clause of a `use` statement, or in the
list of symbols brought in for unqualified access via an `import` statement.

Resolves #17003 

As part of this work, we had to fix a couple of bugs.  Overload sets were
not accounting for operators properly, which caused issues for the IO
operator, `<~>`.  Additionally, the handling of `except` clauses and methods
did not properly adjust for operators (since operators weren't able to be in
`except` clauses), accidentally making it impossible to exclude an operator.
That bug has now been fixed as well.

We also had to remove the parser support for `except *`, which was
deprecated last release.  Its presence in the parser led to reduce/reduce
conflicts, and since it was already deprecated we just removed parser
support for it.  Also removed remaining code in the compiler that handled
it (a case each in UseStmt building and forwarding statement support that
transformed it into `only;` - as a result, no further modifications are
necessary).

Added tests for:
- every operator in an `only` list (as the only symbol, as the first symbol,
and as the last symbol listed)
- every operator in an `import` list surrounded by `{}` (as the only symbol,
as the first symbol, and as the last symbol listed)
- every operator as the last name in an `import` statement (e.g. `import M.+`)
- every operator except `=` in an `except` list (because we shouldn't write `=`
overloads outside the type's defining module)

Removes the tests for deprecating `except *` (which previously enabled
qualified access only for the listed module).

TODO:
- [x] forwarding tests demonstrating we no longer get syntax errors (but I still
expect other errors due to not having merged my operator forwarding support
branch)
- [x] updates for compiler/next
- [x] add tests to compiler/next?
- [x] update to main and fix conflicts
- [x] commit parser changes (for both the old and new compiler)

### CODE REVIEW

**Comment by lydia-duncan** [comment]
check_compiler_next_tests will fail until I merge the next/parser change, I believe

File: test/visibility/except/operatorsExceptions/exceptNegation.compopts [hunk 1]
```diff
@@ -0,0 +1 @@
+DefinesOp.chpl --main-module exceptNegation
```
Comment (mppf) [inline test/visibility/except/operatorsExceptions/exceptNegation.compopts:1]: I think you could make a directory wide compopts with just DefinesOp, right? And you wouldn't need a --main-module if each test has a `main`?

File: test/visibility/import/enablesUnqualified/multipleUnqualified/operatorsFirst.chpl [hunk 2]
```diff
@@ -0,0 +1,558 @@
+module DefinesOp {
```
Comment (mppf) [inline test/visibility/import/enablesUnqualified/multipleUnqualified/operatorsFirst.chpl:1]: Is this different from other `module DefinesOp` test code? Could we have just one copy of it and symbolically link or have COMPOPTS include it in the module search path?

**Review by mppf — APPROVED** [review]
Looks good. Not anything to worry about if it is really hard but it seemed like some duplicate code in the tests could be using the same copy of the code.

File: test/visibility/except/operatorsExceptions/exceptNegation.compopts (hunk 1 above)
Comment (lydia-duncan) [reply test/visibility/except/operatorsExceptions/exceptNegation.compopts:1]: Oh, yeah, good point!

File: test/visibility/import/enablesUnqualified/multipleUnqualified/operatorsFirst.chpl (hunk 2 above)
Comment (lydia-duncan) [reply test/visibility/import/enablesUnqualified/multipleUnqualified/operatorsFirst.chpl:1]: Eh, I think there's value in testing that it works both living inside and outside the file that uses it.  It's probably worth having both strategies for each style of use/import involved, but that also seemed like going a bit too deep on something that so far as I can tell does currently work.

Comment (mppf) [reply test/visibility/import/enablesUnqualified/multipleUnqualified/operatorsFirst.chpl:1]: Would submodule-in-separate-file allow you to have this cake and eat it too?

Comment (lydia-duncan) [reply test/visibility/import/enablesUnqualified/multipleUnqualified/operatorsFirst.chpl:1]: I think of that more as a third way that would warrant testing on its own

**Pull request merged**